src/ethereum/frontier/spec.py

```diff
@@ -692,6 +782,25 @@ def compute_header_hash(header: Header) -> Hash32:
     """
     Computes the hash of a block header.
 
+    The header hash of a block is the canonical hash that is used to refer
+    to a specific block and completely distinguishes a block from another.
+
+    ``keccak256`` is a function that produces a 256 bit hash of any input.
+    It also takes in any number of items as an input and produces a single
+    hash for them. A hash is a completely unique output for a single input.
+    So an input corresponds to one unique hash that can be used to identify
+    the input exactly.
+
+    Prior to using the ``keccak256`` hash function, the header must be
+    encoded using the recursive-length prefix (RLP) serialization format.
+    RLP encoding the header converts it into a space-efficient format that
+    allows for easy transfer of data between nodes. Ethereum.org states, "The
```

Comment:
I think we're fine to not credit ourselves :rofl: 

```suggestion
    allows for easy transfer of data between nodes. The
```